src/ui/molecules/header/Header.tsx

```diff
@@ -1,30 +1,119 @@
-import React from 'react'
-import './header.scss'
-import { ThemeSwitcher } from 'ui/atoms/theme/ThemeSwitcher'
-import type { DeepReadonly } from 'superTypes'
-import { useBreakpoint } from 'hook/useBreakpoint'
-import type { Breakpoints } from 'hook/useBreakpoint'
+import React, { useCallback, useState } from 'react'
+import type { DeepReadonly, UseState } from 'superTypes'
 import classNames from 'classnames'
+import { useMediaType } from 'hook/useMediaType'
+import { ThemeSwitcher as Switcher } from 'ui/atoms/theme/ThemeSwitcher'
+import { Icon } from 'ui/atoms/icon/Icon'
+import './header.scss'
 
 export type HeaderProps = {
   /**
    * Element positioned first in the reading flow.
    */
   readonly firstElement: JSX.Element
+  /**
+   * The list of navigable links that make up the menu.
+   */
+  readonly navigationMenu?: JSX.Element[]
 }
 
+const BurgerMenu = ({
+  isOpened,
+  onToggle
+}: DeepReadonly<{ isOpened: boolean; onToggle: () => void }>): JSX.Element => (
+  <div
+    className={classNames(
+      'okp4-header-navigation-burger-menu',
+      isOpened ? 'rotate-down' : 'rotate-up'
+    )}
+    onClick={onToggle}
+  >
+    <Icon name="menu" size={38} />
+  </div>
+)
+
+const BurgerMenuList = ({
+  navigation
+}: DeepReadonly<{ navigation: JSX.Element[] }>): JSX.Element => (
+  <div className="okp4-header-navigation-burger-menu-list">
+    {navigation.map((link: Readonly<JSX.Element>, index: number) => (
+      <div className="okp4-header-navigation-burger-menu-item" key={index}>
+        {link}
+      </div>
+    ))}
+  </div>
+)
+
+const RowMenuList = ({ navigation }: DeepReadonly<{ navigation: JSX.Element[] }>): JSX.Element => (
```

Comment:
I think we could gather the `BurgerMenuList` and the `RowMenuList` in one single component as they do the same thing, except the classNames that can be declared conditionally.

Reply:
Yes, good idea !